File: c/game.c
#include<stdio.h>
#include<stdlib.h>
#include<string.h>
#include<time.h>

int
main(void) {
    int i, random_num, guess_count, guess_int;
    char c, guess_str[1024];

    puts("Welcome to the number guessing game imlemented in C.");
    puts("Please guess a number that has been randomly chosen between 1 and 100.");
    puts("You only have 9 guesses, so make them count!");
    puts("Type 'quit' as your guess at any time to quit the game.");

    srand(time(0));
    random_num = (rand() % 100) + 1;

    guess_count = 1;
    while (guess_count < 10) {
        // Prompt user for a guess.
        printf("\nGuess number %d: ", guess_count);

        // Get user input.
        i = 0;
        while ((c = getchar()) != '\n') {
            guess_str[i] = c;
            i++;
        }
        guess_str[i] = '\0';

        // Convert into int
        guess_int = atoi(guess_str);

        // Handle user input.
        if (strcmp(guess_str, "quit") == 0) {
            puts("Bye!");
            guess_count = 11;
        } else if (guess_int == 0) {
            if (strcmp(guess_str, "0") == 0) {
                puts("Please guess a number between 1 and 100.");
            } else {
                puts("This is not a valid input!");
            }
        } else {
            if (guess_int == random_num) {
                puts("Congradulations! You guessed the correct number!");
                guess_count = 11;
            } else if (guess_int < 0 || 100 < guess_int) {
                puts("Please guess a number between 1 and 100.");
            } else if (guess_int > random_num) {
                puts("Your guess is too high, try again!");
                guess_count++;
            } else if (guess_int < random_num) {
                puts("Your guess is too low, try again!");
                guess_count++;
            }
        }

    }

    if (guess_count == 10) {
        puts("\nWait... you ran out of guesses! That's it, you lose!");
    }

    return 0;
}

File: go/game.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"math/rand"
	"os"
	"strconv"
	"strings"
)

func main() {
	reader := bufio.NewReader(os.Stdin)

	fmt.Println("Welcome to the number guessing game imlemented in C.")
	fmt.Println("Please guess a number that has been randomly chosen between 1 and 100.")
	fmt.Println("You only have 9 guesses, so make them count!")
	fmt.Println("Type 'quit' as your guess at any time to quit the game.")

	randomNum := rand.Intn(100) + 1

	guessCount := 1
	for guessCount < 10 {
		// Prompt user for a guess.
		fmt.Printf("\nGuess number %d: ", guessCount)

		// Get user input
		guessStr, err := reader.ReadString('\n')
		if err != nil {
			log.Fatal(err)
		}
		guessStr = strings.Trim(guessStr, "\n")

		// Handle user input
		if guessStr == "quit" {
			fmt.Println("Bye!")
			break
		}

		guessInt, err := strconv.Atoi(guessStr)
		if err != nil {
			fmt.Println("This is not a valid input!")
			continue
		}

		if guessInt == randomNum {
			fmt.Println("Congradulations! You guessed the correct number!")
			break

		} else if guessInt < 1 || guessInt > 100 {
			fmt.Println("Please guess a number between 1 and 100.")
			continue

		} else if guessInt > randomNum {
			fmt.Println("Your guess is too high, try again!")
			guessCount++
			continue

		} else if guessInt < randomNum {
			fmt.Println("You guess is too low, try again!")
			guessCount++
			continue

		}
	}

	// Only if you ran out of guesses will the count be 10
	if guessCount == 10 {
		fmt.Println("\n Wait... you ran out of guesses! That's it, you lose!")
	}
}
